File: frontend/src/lib/test-shared-types.ts
// Test file to validate cross-workspace imports from @jornada/shared
import { AssessmentSchema, UserSchema } from '@jornada/shared';

// This file validates that frontend can import types from shared workspace
export function testSharedTypes() {
  const user: UserSchema = {
    id: '1',
    email: 'test@example.com',
    full_name: 'Test User',
    role: 'user'
  };

  const assessment: AssessmentSchema = {
    id: '1',
    user_id: user.id,
    status: 'draft',
    scores: {}
  };

  return { user, assessment };
}

File: backend/test_shared_types.py
"""
Test file to validate that backend can reference shared types.

Note: Python doesn't directly import TypeScript types, but this validates
the workspace structure and that shared types are accessible for documentation
and validation purposes.
"""

# For Python, we would typically use Pydantic models that mirror the TypeScript types
# This is a placeholder to demonstrate the concept

def test_shared_types_reference():
    """
    Validates that the shared types workspace exists and is accessible.
    In production, we would use Pydantic models that match the TypeScript schemas.
    """
    import os
    shared_path = os.path.join(os.path.dirname(__file__), '..', 'shared')
    assert os.path.exists(shared_path), "Shared workspace should exist"

    user_types_path = os.path.join(shared_path, 'types', 'user.ts')
    assessment_types_path = os.path.join(shared_path, 'types', 'assessment.ts')

    assert os.path.exists(user_types_path), "User types should exist"
    assert os.path.exists(assessment_types_path), "Assessment types should exist"

    print("✅ Shared types workspace validated")
    return True

if __name__ == "__main__":
    test_shared_types_reference()
